deployer/src/deployment/run.rs

```diff
@@ -470,33 +514,36 @@ mod tests {
             _ = sleep(Duration::from_secs(5)) => panic!("cleanup should have been called as service stopped on its own"),
             Ok(()) = cleanup_recv => {},
         }
+
+        // Prevent the runtime manager from dropping earlier, which will kill the processes it manages
+        drop(runtime_manager);
```

Comment:
Clever :brain:

Reply:
Took me a while to figure why the test kept failing :sweat_smile:

Reply:
I can imagine! :smile: